api_v3/services/SessionService.php

```diff
@@ -294,10 +295,18 @@ function startWidgetSession ( $widgetId , $expiry = 86400 )
 		// according to the partner's policy and the widget's policy - define the privileges of the ks
 		// TODO - decide !! - for now only view - any kshow
 		$privileges = "view:*,widget:1";
-		
-		if(PermissionPeer::isValidForPartner(PermissionName::FEATURE_ENTITLEMENT, $partnerId) &&
+
+		if (PermissionPeer::isValidForPartner(PermissionName::FEATURE_ENTITLEMENT, $partnerId) &&
 			!$widget->getEnforceEntitlement() && $widget->getEntryId())
-			$privileges .= ','. kSessionBase::PRIVILEGE_DISABLE_ENTITLEMENT_FOR_ENTRY . ':' . $widget->getEntryId();
+		{
+			$entryId = $widget->getEntryId();
+			$privileges .= ',' . kSessionBase::PRIVILEGE_DISABLE_ENTITLEMENT_FOR_ENTRY . ':' . $entryId;
+			$entry = entryPeer::retrieveByPKNoFilter($entryId, null, false);
```

Comment:
If this widget is not allowed to access entry , then why are you calling - retrieveByPKNoFilter (use filter!)
This can be a security breach

Reply:
The widget allows accessing the entry and disables the entitlement for it, we cant use filter in this context cause you don't use ks to generate the widget